colossalai/gemini/gemini_mgr.py

```diff
@@ -57,22 +62,19 @@ def adjust_layout(self, chunks: Tuple[Chunk, ...], group_name: str) -> None:
                                                                warmup=self._warmup,
                                                                compute_list=self._compute_list,
                                                                compute_idx=self._compute_idx)
-        self._cpu_gpu_move_volume += vol
+        self._d2h_volume += vol
         self._evict_time += evict_time
         # move COMPUTE tensors to CUDA
-        self._cpu_gpu_move_volume += cuda_demand
+        self._h2d_volume += cuda_demand
 
-    @property
-    def cpu_gpu_move_volume(self):
-        return self._cpu_gpu_move_volume
-
-    # @functools.lru_cache(maxsize=None)
-    # TODO: test lru
+    @functools.lru_cache(maxsize=None)
```

Comment:
why add this decorator?
I suppose it works for a recursion function.

Reply:
layout info at each computation step won’t change during steady phase. use LRU to speed up this function